morepath/app.py

```diff
@@ -255,17 +247,176 @@ def set_setting_section(section, section_settings):
         for section, section_settings in settings.items():
             set_setting_section(section, section_settings)
 
-    def _get_class_path(self, model, variables):
-        """Path for a model class and variables.
+    @reg.dispatch_method(get_key_lookup=cached_key_lookup)
+    def _view(self, obj, request):
```

Comment:
I can't make it public as ``view`` as we already have a directive with that name. I will call it ``get_view``.